sections/sprint-4/updated-uppaal.tex

```diff
@@ -0,0 +1,144 @@
+\section{Updated UPPAL}
+As mentioned in \autoref{sec:update-network-protocol} the approach to handling the network data has changed from being purely UDP to a combination of UDP and TCP.
+This lead to the previous \uppaal model specified in \autoref{sec:sprint3-uppaal} being outdated and that it had to be updated to fit the new requirements.
+\\
+Additionally, the previous model focused mostly on the flow of the program and did not properly take the network part of the system into account.
+In this new model, the system will consist of four parts: The host, which is meant to represent the Python program that the game facilitator uses, an arbitrary number of clients who have connected to the game, and the UDP and TCP connection between those.
+
+\subsection{System declarations}
+\begin{uppaalcode}[caption={System declarations}, label={lst:uppaal4:systemdecl}]
+    system UDP_host, TCP_host, TCP_client, Host, Client;
+\end{uppaalcode}
+Before diving into the details of the different templates, let us first get an overview of the system:\\
+The project consists of five templates: \uppTemp{Host}, \uppTemp{Client}, \uppTemp{TCP_client}, \uppTemp{TCP_host}, \uppTemp{UDP_host}.
+As mentioned above, the \uppTemp{Host} and \uppTemp{Client} templates are meant to represent the interface that users interact with when using the system.
+In this new version of the model, the inclusion of TCP clients and hosts, and a UDP host additionally allows us to model the behavior of the network aspect of the system.
+
+% System declerations
+\subsection{Global declarations}
+The global declarations (seen on \autoref{lst:uppaal4:globaldecl}) for the system have two purposes: Setting game and simulation specific options, such as how many clients should be instantiated, or how long the TCP client should wait for a response before declaring the connection as timeouted.
+The \uppVar{host} variable is saved as an alias to know which space in the buffers is reserved for messages from clients to the host.
+To facilitate data transfer between the templates, the model makes use of global buffers, which are defined as int arrays.
+Each place in the array corresponds to a buffer related to a client process so that \uppProc(Client(0)) will read from the 0th index of the array, \uppProc(Client(1)) will read from the second and so on.
```

Comment:
```suggestion
Each place in the array corresponds to a buffer related to a client process so that \uppProc(Client(0)) will read from index 0 of the array, \uppProc(Client(1)) will read from index 1 and so on.
```